src/utils/nullishUtils.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2023 PixieBrix, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+export type Nullish = null | undefined;
+
+export type Nullishable<T> = T | Nullish;
+
+// Discussion: Do we even want to allow Optional? Or should we enforce Nullishable only?
+export type Optional<T> = T | undefined;
+
+// Discussion: Do we even want to allow Nullable? Or should we enforce Nullishable only?
+export type Nullable<T> = T | null;
+
+// TODO: create lint rule to enforce use of NonNullish over NonNullable
+export type NonNullish<T> = Exclude<T, Nullish>;
```

Comment:
If this must exist, then it can just be 
```suggestion
export type NonNullish<T> = NonNullable<T>;
```

But like with `isNullish` IMHO it's not particularly productive to rename, discuss, enforce new types over something that exists and works. ESLint isn't a silver bullet https://xkcd.com/1319/

Reply:
I think I originally added that type because we were discussing having `Nullable<T>` and `Optional<T>` in addition to `Nullishable<T>`, and the behavior would clash with certain assumptions given to us by `NonNullable`. I think the better option is to just exclude `Nullable<T>` and `Optional<T>`.